### ISSUE TITLE
Finger joints not being created on one of the two components

### ISSUE THREAD

**Issue by ExR90** [FlorianPommerening/FingerJoints#19]
Hello,

The tool has worked great on other projects. I cannot fathom why it is not working now. I have a series of components that all joint into a main body. The first one works fine, the subsequent ones do NOT. These are unique copies, not related iterations. I had this same issue earlier in the day and deleted/recreated a few times and it worked. I have tried the same things this time and cannot get through this. Any ideas?

Video of the behavior:

[https://share.getcloudapp.com/eDu7nkJ4](https://share.getcloudapp.com/eDu7nkJ4)

**Comment by ExR90** [comment]
I have since figured out how to get around this, for whatever reason I have to "Capture Position" after making the copy - then the finger joint will work correctly. I don't think that is expected behavior but at least I have a way around the issue.

**Comment by FlorianPommerening** [comment]
Good to hear you got it working. The feature is position dependent, which means that you have to capture the position of objects before using the feature. Another way to approach this would be to cut the joint into the first piece before copying it and then use a combine feature (set to "cut" and "keep tools") on the copies and the back board to cut the holes in the back board.

Could you attach an example file here at the point in the timeline where the unexpected behavior happens? I can try if it is possible to at least print a warning when this situation occurs.

**Comment by ExR90** [comment]
The mentioned alternate way to do it with "instanced" copies doesn't work unless I'm not understanding. Jointing one joints all the copies, but only creates holes on the target where the first (selected) component intersected. It does not create the matching holes on the target body for the copies. If you try to joint the copies, you get an error - see below for screenshot.

See attached demo file. Reproduce Steps:

  1.  Create a box, convert to component.
  2.  Create another box, convert to component.
  3.  Arrange the objects to overlap as if you were going to joint them
  4.  Select one of the objects, copy. Then "Paste New" (not paste).
  5.  Move it elsewhere on the first box, aligned for jointing.
  6.  Joint object 1, note the effect.
  7.  Joint object 2, note the different effect.

Now, to reproduce the error mentioned, continue by doing:

  1.  Roll back the timeline to right before you made the copy.
  2.  Copy/paste it but this time use "Paste" not "Paste New".
  3.  Move the new object to align for jointing.
  4.  Joint any one of these objects. It will joint them all but target body will only get holes made for the one selected object, not the instanced copies.
  5.  Try to joint the remaining items and it will throw an error. This leaves only capturing the position before jointing AND they must be unique. Instanced copies cause it to generate that error EVEN IF you capture position.

Solution: Capture Positions after copying/pasting, before jointing and do so with unique copies, not instanced copies tied to a master item.

[cid:21733852-46a1-48b3-a648-fe8a31dde56f]

________________________________

**Comment by FlorianPommerening** [comment]
I think the attachment didn't work (if you are replying by email, Github might not forward the attachment).

What you describe is not the workflow I had in mind as an alternative. What I meant was to use the joint operation only once on the original piece (never on the copy). Then this will create the "finger" in all copies but only one hole in the back board. But the fingers of all copies will overlap the backboard, so you can cut them out with a single combine operation (the one built into Fusion). For that workflow, both "paste new" and "paste" should work. The downside is that it does not support the experimental gaps between the fingers and the holes.

**Comment by FlorianPommerening** [comment]
I'll close the issue, as it seems the problem is solved for now. If you disagree, feel free to reopen it or open a follow-up issue.
